### PR TITLE
Upgrade internal packages to support ESLint >= 6

### LINKED ISSUE #28293: [eslint-config-react-native-community] Upgrade @typescript-eslint/eslint-plugin
Description:

 Same as #28214 but with environment info to satisfy bots. I'm not sure why the original poster closed his issue, but upgrading eslint-config to latest plugins would be nice, if only for the parity of packages and to remove install warnings.

React Native version:

```
System:
    OS: Linux 5.3 Ubuntu 19.10 (Eoan Ermine)
    CPU: (4) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    Memory: 7.52 GB / 19.43 GB
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
    Yarn: Not Found
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.1/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce

1. Install `react-native-community/eslint-config` on a fresh RN project
2. Result:

```
npm WARN @typescript-eslint/eslint-plugin@1.13.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @typescript-eslint/parser@1.13.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
```

## Expected Results

A fresh RN project uses ESLint 6, so it stands to reason that the eslint-config package should be compatible.

### PR DESCRIPTION
**Pull request by Barbiero**
Fixes #28293

I've tested it with https://github.com/react-native-community/react-native-template-typescript and it seems to be working as expected - no warnings, supports typescript 3.8. 

(note: I didn't upgrade the package version as I don't know how the releases work for this package)

## Changelog

[CATEGORY] [TYPE] - Message

### CODE REVIEW

**Comment by Barbiero** [comment]
> Hi @Barbiero!
> 
> Thank you for your pull request and welcome to our community.We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file.
> 
> In order for us to review and merge your code, please sign at https://code.facebook.com/cla. **If you are contributing on behalf of someone else (eg your employer)**, the individual CLA may not be sufficient and your employer may need to sign the corporate CLA.
> 
> If you have received this in error or have any questions, please contact us at [cla@fb.com](mailto:cla@fb.com?subject=CLA%20for%20facebook%2Freact-native%20%2328393). Thanks!

done

File: packages/eslint-config-react-native-community/package.json [hunk 1]
```diff
@@ -10,18 +10,18 @@
...
-    "eslint-plugin-react-hooks": "^2.0.1",
+    "eslint-plugin-prettier": "3.1.2",
+    "eslint-plugin-react": "7.19.0",
+    "eslint-plugin-react-hooks": "^2.5.1",
     "eslint-plugin-react-native": "3.8.1",
-    "prettier": "1.17.0"
+    "prettier": "^2.0.2"
   },
   "peerDependencies": {
     "eslint": ">=5"
```
Comment (matt-oakes) [inline packages/eslint-config-react-native-community/package.json]: Does this need to be updated as well, or is ESLint 5 still supported?

Comment (Barbiero) [reply packages/eslint-config-react-native-community/package.json]: from my tests, it seems eslint >=6 is required. Thanks for that

**Comment by matt-oakes** [comment]
I've just published this to NPM as v1.0.0. You should be able to use it now 👍

**Pull request closed without merging**